### PR TITLE
feat(deno/sdk): native function embedding in typescript

### PR DESCRIPTION
**Pull request by afmika**
Add support for function or lambda definition typescript sdk for `deno.func` similarly to how `python.from_def` in python sdk works.

#### Motivation and context

Providing a string is a bit impractical and counter-intuitive espcially when the sdk language matches with runtime's language.

#### Migration notes

No changes needed.

### Checklist

- [x] The change come with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

**Review by zifeo — APPROVED** [review]

**Review by zifeo — APPROVED** [review]

File: typegraph/node/sdk/src/runtimes/deno.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ interface PredefinedFuncMat extends Materializer {
 }
 
 export interface DenoFunc {
-  code: string;
+  code: string | Function;
```
Comment (Natoandro) [inline typegraph/node/sdk/src/runtimes/deno.ts:29]: The name `code` would suggest that it is a string...

File: typegraph/node/sdk/src/runtimes/deno.ts [hunk 2]
```diff
@@ -38,6 +38,26 @@ export interface DenoImport {
   effect?: Effect;
 }
 
+function stringifyFn(code: string | Function) {
+  if (typeof code == "function") {
+    const source = code.toString();
+    const namedFnMatch = source.match(/function\s*(\*?\s*[a-zA-Z0-9_]+)/);
+    if (namedFnMatch) {
+      const [, name] = namedFnMatch;
+      if (name.replace(/s/g, "").startsWith("*")) {
```
Comment (Natoandro) [inline typegraph/node/sdk/src/runtimes/deno.ts:47]: Do you mean character `'s'`? Or perhaps `\s`?

Comment (afmika) [reply typegraph/node/sdk/src/runtimes/deno.ts:47]: Nice catch!

File: typegraph/node/sdk/src/runtimes/deno.ts (hunk 1 above)
Comment (afmika) [reply typegraph/node/sdk/src/runtimes/deno.ts:29]: Yeah. right. I thought of renaming it but it will break old typegraphs that uses the deno feat.

**Review by Natoandro — APPROVED** [review]

**Pull request merged**